src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -67,7 +68,7 @@ public function onMessageFailed(WorkerMessageFailedEvent $event)
             }
 
             // add the delay and retry stamp info
-            $retryEnvelope = $this->withLimitedHistory($envelope, new DelayStamp($delay), new RedeliveryStamp($retryCount));
+            $retryEnvelope = $this->withLimitedHistory($envelope, new DelayStamp($delay), new RedeliveryStamp($retryCount, $throwable->getMessage(), FlattenException::createFromThrowable($throwable)));
```

Comment:
@fabpot @weaverryan This will avoid the FlattenException to be null when retrying, in order to accommodate the changes on FlattenExceptionNormalizer.